src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -217,18 +217,16 @@ public void init() {
      * @throws IllegalStateException when validation of default javadoc tokens fails
      */
     private void validateDefaultJavadocTokens() {
-        if (getRequiredJavadocTokens().length != 0) {
-            final int[] defaultJavadocTokens = getDefaultJavadocTokens();
-            Arrays.sort(defaultJavadocTokens);
-            for (final int javadocToken : getRequiredJavadocTokens()) {
-                if (Arrays.binarySearch(defaultJavadocTokens, javadocToken) < 0) {
-                    final String message = String.format(Locale.ROOT,
-                            "Javadoc Token \"%s\" from required javadoc "
-                                + "tokens was not found in default "
-                                + "javadoc tokens list in check %s",
-                            javadocToken, getClass().getName());
-                    throw new IllegalStateException(message);
-                }
+        final int[] defaultJavadocTokens = getDefaultJavadocTokens();
+        Arrays.sort(defaultJavadocTokens);
+        for (final int javadocToken : getRequiredJavadocTokens()) {
+            if (Arrays.binarySearch(defaultJavadocTokens, javadocToken) < 0) {
+                final String message = String.format(Locale.ROOT,
+                        "Javadoc Token \"%s\" from required javadoc "
+                            + "tokens was not found in default "
+                            + "javadoc tokens list in check %s",
+                        javadocToken, getClass().getName());
+                throw new IllegalStateException(message);
             }
         }
```

Comment:
The existing code in this method is crazy, we need to clean this up. Binary search and sorting of default tokens is overkill for such a simple task. We can just do something like this here:
```suggestion
        final Set<Integer> defaultTokens = Arrays.stream(getDefaultJavadocTokens())
                .boxed()
                .collect(Collectors.toUnmodifiableSet());

        final List<Integer> missingRequiredTokens = Arrays.stream(getRequiredJavadocTokens())
                .boxed()
                .filter(token -> !defaultTokens.contains(token))
                .collect(Collectors.toUnmodifiableList());

        missingRequiredTokens.forEach(token -> {
            final String message = String.format(Locale.ROOT,
                    "Javadoc Token \"%s\" from required javadoc"
                            + "tokens was not found in default "
                        + "javadoc tokens list in check %s",
                    JavadocUtil.getTokenName(token), getClass().getName());
            throw new IllegalStateException(message);
        });
```

IMO this is much easier to read and understand, provides a much more helpful exception message, avoids mutable collections, and should kill the mutation.